_source/_posts/2022-05-04-angular-microfrontend-deploy.md

```diff
@@ -0,0 +1,286 @@
+
+### Add your deployment URL to Okta
+
+You'll need to update the Okta application with the new deploy location so that you can sign in. Open the Okta dashboard. Navigate to **Applications** > **Applications** and select the Okta application you created for this project. On the **General** tab, press the **Edit** button on the **General Settings** section and add the new deploy URI for the `shell` application.
+
+{% img blog/angular-microfrontend-deploy/okta-configure-redirects.jpg alt:"Okta configuration for adding the Vercel deploy URI of shell application as 'Sign-in redirect URIs' and 'Sign-out redirect URIs'. The 'Sign-in redirect URI' should include the '/login/callback' route, while the 'Sign-out redirect URIs' is the deploy URI as is." width:"800" %}{: .center-image }
+
+Next, add the deploy URI as a Trusted Origin in Okta. Navigate to **Security** > **API** and then navigate to the **Trusted Origins** tab. Press **+ Add Origin**. Add the deploy location and select **CORS** for the type.
+
+{% img blog/angular-microfrontend-deploy/okta-configure-trusted-origin.jpg alt:"Okta configuration for adding the Vercel deploy URI of shell application to trusted origin. Type 'CORS' is selected." width:"800" %}{: .center-image }
+
+Commit your changes and push to your main branch. After the build succeeds, you should be able to use the deployed site from end to end!
+
+
+## Secure your micro-frontends
+
+Right now, you can access what should be protected routes by manually typing in the full URI to the profile route. Even though you don't see any profile information, we aren't guarding the route. Let's make the site a little more secure by protecting the route.
+
+Open `projects/shell/src/app/app-routing.module.ts`. We'll add a route guard that comes out of the box with Okta's Angular SDK. Update the route definition for the `profile` route as shown below.
+
+```ts
+import { OktaCallbackComponent, OktaAuthGuard} from '@okta/okta-angular';
+
+{
+  path: 'profile',
+  loadChildren: () => loadRemoteModule({
+    type: 'module',
+    remoteEntry: `${environment.mfe.mfeProfile}/remoteEntry.js`,
+    exposedModule: './Module'
+  }).then(m => m.ProfileModule),
+  canActivate: [OktaAuthGuard]
+},
+```
+
+Now what happens is if you navigate directly to the profile route, you'll first redirect to Okta's sign-in page, then redirect back to the profile route. 
+
+Route protection works well from within the micro-frontend shell. But what if you navigate directly to the URI where the profile micro-frontend resides? You will still be able to navigate to the profile route within the micro-frontend because there isn't a guard within the profile application. 
```

Comment:
```suggestion
Route protection works well from within the micro-frontend shell. But what if you navigate directly to the URI where the profile micro frontend resides? You will still be able to navigate to the profile route within the micro frontend because there isn't a guard within the profile application. 
```
two words, no hyphen